### PR TITLE
test: move unit tests to use node:test

### PR DESCRIPTION
**Pull request by ematipico**
## Changes

Part of https://github.com/withastro/astro/issues/9873

This PR ports ALL unit tests to use `node:test`.

## Testing

CI should pass

### CODE REVIEW

File: packages/astro/test/units/routing/route-sanitization.test.js [hunk 1]
```diff
@@ -5,7 +5,8 @@ import {
 	defaultLogger,
```
Comment (Princesseuh) [inline packages/astro/test/units/routing/route-sanitization.test.js:5]: File wasn't renamed

File: packages/astro/test/units/routing/trailing-slash.test.js [hunk 2]
```diff
@@ -5,7 +5,8 @@ import {
 	defaultLogger,
```
Comment (Princesseuh) [inline packages/astro/test/units/routing/trailing-slash.test.js:5]: File wasn't renamed

File: packages/astro/test/units/vite-plugin-astro-server/response.test.js [hunk 3]
```diff
@@ -5,7 +5,8 @@ import {
 	defaultLogger,
```
Comment (Princesseuh) [inline packages/astro/test/units/vite-plugin-astro-server/response.test.js:5]: File wasn't renamed

**Review by Princesseuh — APPROVED** [review]
I'm not quite sure why some of the files didn't get recognized by Git as being moves, weird. It looks great to me though!

**Comment by ematipico** [comment]
> I'm not quite sure why some of the files didn't get recognized by Git as being moves, weird. It looks great to me though!

Just a guess, but during the rename, I didn't actually rename a file, but made a copy and deleted the old one.

**Pull request merged**